armi/materials/material.py

```diff
@@ -101,6 +122,26 @@ def getChildrenWithFlags(self, typeSpec: TypeSpec, exactMatch=True):
         """Return empty list, representing that this object has no children."""
         return []
 
+    def backUp(self):
+        """Empty because statelss materials shouldn't use cache"""
+        pass
+
+    def restoreBackup(self, paramsToApply):
+        """Empty because statelss materials shouldn't use cache"""
+        pass
+
+    def clearCache(self):
+        """Empty because statelss materials shouldn't use cache"""
+        pass
+
+    def _getCached(self, name):
+        """Empty because statelss materials shouldn't use cache"""
+        return None
+
+    def _setCache(self, name, val):
+        """Empty because statelss materials shouldn't use cache"""
+        pass
```

Comment:
Can we just delete all these?

Spelling error: `stateless`

Big picture: have you measured the performance impact of not caching the material properties vs. temperature? That was a hugely major performance bottleneck we identified through formal profiling of test cases. 

We need to see a profiler trace of loading a reactor and printing out all number densities in order to judge this. Can you paste the before/after highlights of the profiler results as a comment to this PR?

Reply:
Two questions:

1. Isn't a cache a "state" for the class?
2. In downstream repos, we have temperature-dependent densities.  These caches will preserve the first value they get forever. Shouldn't that lead to incorrect material densities?

Reply:
I will look into this more once I have fully removed the Parameters from `Material`s.

Reply:
Done.

Since `Material` no longer subclasses `Composite`, I had to mostly copy them over from the `Composite` class.  But now they aren't blank here any more.